src/acceleration/IQNIMVJAcceleration.cpp

```diff
@@ -79,45 +80,6 @@ void IQNIMVJAcceleration::initialize(
 
   if (_imvjRestartType > 0)
     _imvjRestart = true;
-
-  int entries        = _primaryResiduals.size();
-  int cplDataEntries = _residuals.size();
-  int global_n       = 0;
-
-  if (!utils::IntraComm::isParallel()) {
-    global_n = cplDataEntries;
-  } else {
-    global_n = _dimOffsets.back();
-  }
-
-  // initialize parallel matrix-matrix operation module
-  _parMatrixOps = std::make_shared<impl::ParallelMatrixOperations>();
-  _parMatrixOps->initialize(not _imvjRestart);
-  _svdJ.initialize(_parMatrixOps, global_n, getLSSystemRows());
-
-  if (not _imvjRestart) {
-    // only need memory for Jacobain if not in restart mode
-    _invJacobian    = Eigen::MatrixXd::Zero(global_n, entries);
-    _oldInvJacobian = Eigen::MatrixXd::Zero(global_n, entries);
-  }
-
-  // initialize parallel matrix-matrix operation module
-  _parMatrixOps = std::make_shared<impl::ParallelMatrixOperations>();
-  _parMatrixOps->initialize(not _imvjRestart);
-  _svdJ.initialize(_parMatrixOps, global_n, getLSSystemRows());
-
-  // initialize V, W matrices for the LS restart
-  if (_imvjRestartType == RS_LS) {
-    _matrixCols_RSLS.push_front(0);
-    _matrixV_RSLS = Eigen::MatrixXd::Zero(entries, 0);
-    _matrixW_RSLS = Eigen::MatrixXd::Zero(cplDataEntries, 0);
-  }
-  _Wtil = Eigen::MatrixXd::Zero(cplDataEntries, 0);
-
-  if (utils::IntraComm::isPrimary() || !utils::IntraComm::isParallel()) {
-    _infostringstream << " IMVJ restart mode: " << _imvjRestart << "\n chunk size: " << _chunkSize << "\n trunc eps: " << _svdJ.getThreshold() << "\n R_RS: " << _RSLSreusedTimeWindows << "\n--------\n"
-                      << '\n';
-  }
```

Comment:
Can you explain why all this code could be removed? Did we decide to delete an (unused) feature?

Edit: Just saw that you moved it into `specializedInitializeVectorsAndPreconditioner`. Again: Is this maybe a refactoring we could directly merge into develop?

Reply:
Yes, I had to move this code into specializedInitializeVectorsAndPreconditioner. The reason behind this change is that the vector and preconditioner initialization needs to know the time grid of the Quasi-Newton method, which is determined after the first waveform iteration. 

It probably makes sense to refactor this change in interpolation in a separate pull request. It would also be nice if there was a way of making this more robust for future changes and easier to follow.